wfl/utils/save_calc_results.py

```diff
@@ -113,6 +131,34 @@ def save_calc_results(atoms, *, prefix, properties):
             atoms.new_array(prefix + p, v)
 
 
+def is_per_atom(prop, num_atoms):
+    """Tries to guess whether a property is per-atom.
+
+    Parameters
+    ----------
+    prop: int/float/np.ndarray
+        Property to check
+    num_atoms:
+        Number of atoms in the structure
+
+    Returns 
+    -------
+    True if shape of the property matches number of atoms 
+    """
+
+    if isinstance(prop, (int, float)):
+        return False
+
+    if num_atoms in [1, 3]:
```

Comment:
I _really_ don't like this.  This is why ASE has problems with arbitrary properties.

Reply:
I would prefer that, if it's not _too_ cumbersome for us